WF_Proton_Launcher/Main.cpp

```diff
@@ -147,6 +150,44 @@ int main(int argc, char** argv)
 	Hashlist hashes;
 	std::vector<std::string> diff;
 
+	if (firstrun) {
+		std::cout << std::endl << "Installing Direct X...";
+
+		FILE* fp;
+		fopen_s(&fp, "directx_Jun2010_redist.exe", "wb");
+		if (!fp)
+		{
+			std::wcout << "Failed to create file on disk." << std::endl;
+		}
```

Comment:
I feel like there's a `return 1;` missing in here, the download is guaranteed to fail if the file wasn't opened after all, and it's not going to be nearly as clean of an error in that case.